go-controller/hybrid-overlay/pkg/controller/master.go

```diff
@@ -284,6 +295,64 @@ func (m *MasterController) DeleteNode(node *kapi.Node) error {
 	if _, ok := node.Annotations[types.HybridOverlayDRMAC]; ok && !houtil.IsHybridOverlayNode(node) {
 		m.deleteOverlayPort(node)
 	}
+
+	if err := removeHybridLRPSharedGW(node.Name); err != nil {
+		return err
+	}
 	klog.V(5).Infof("Node delete for %s completed", node.Name)
 	return nil
 }
+
+func setupHybridLRPSharedGw(nodeSubnets []*net.IPNet, nodeName string) error {
+	klog.Infof("Setting up logical route policy for hybrid subnet on node: %s", nodeName)
+	var L3Prefix string
+	for _, nodeSubnet := range nodeSubnets {
+		if utilnet.IsIPv6CIDR(nodeSubnet) {
+			L3Prefix = "ip6"
+		} else {
+			L3Prefix = "ip4"
+		}
+		var hybridCIDR *net.IPNet
+		for _, hybridSubnet := range config.HybridOverlay.ClusterSubnets {
+			if utilnet.IsIPv6CIDR(hybridSubnet.CIDR) == utilnet.IsIPv6CIDR(nodeSubnet) {
+				hybridCIDR = hybridSubnet.CIDR
+				break
+			}
+		}
+
+		drIP := util.GetNodeHybridOverlayIfAddr(nodeSubnet).IP
+		matchStr := fmt.Sprintf(`inport == "%s%s" && %s.dst == %s /* %s%s */`,
```

Comment:
Could you set the `external_id` on the logical router policy object instead of this comment? It will make syncing these a lot easier than parsing the comments later on

Reply:
+1

Reply:
good question. This is a pattern used for the other route policies which has bit us in the past because of the nuances in regexp matching:
https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/gateway_cleanup.go#L110

@girishmg is there any particular reason we use the comment in the routes there rather than external_ids?

Reply:
Yes, I know it's a legacy pattern we've used to either simplify our setup of logical router policies by using `lr-policy-add` - which does not support setting the external ID - as supposed to `create`, which does but is more "contrived". However, I am working on a patch fixing a bug related to this, and syncing state with an identifier baked in the match condition is not recommended.

Reply:
OK as long as there is no objection from @girishmg I'll make the change

Reply:
Since this is in HO code, it is fine.  If we need to cleanup the other PBRs too, then we will need an upgrade path so I would not touch those policies as it will break cluster upon upgrade to this PR. Thanks.

Reply:
Actually I just realized this is going to be a lot of work. go-ovn bindings dont have support for logical router policy, which means I would have to:

1. output = ovn-nbctl find logical_router_policy external_ids=<something>
2. if len(output)> 0; unmarshal(output) -> some struct that maps all of the fields of a logical router policy
3. compare all of the fields to what I want to add to see if anything needs updating
4. If needs updating, ovn-nbctl remove <entry>
5. ovn-nbctl create logical_router_policy
6. ovn-nbctl add logical_router...

steps 2,3 are a lot of work to add something we are going to throw away for go-ovn bindings...and we dont have the time to wait on go-ovn. The only reasonable solution I can see is:
1. output=ovn-nbctl find logical_router_policy <exact match>
2. if len(output) > 0, do nothing and return; else output=ovn-nbctl find logical_router_policy external_ids=<something>
3. if len(output)> 0; delete policy
4. create policy
5. set to logical router

Will implement it like this...